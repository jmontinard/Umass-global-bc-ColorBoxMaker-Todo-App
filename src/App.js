import React, { useState } from 'react';
import BoxList from './BoxList';
import TodoList from './TodoList';
import './style.scss'; // Assuming you have a separate CSS file for styles

function App() {
  const [showBoxMaker, setShowBoxMaker] = useState(true);

  const toggleApp = () => {
    setShowBoxMaker(!showBoxMaker);
  };

  return (
    <div className="App">
      <button onClick={toggleApp}>
        Switch to {showBoxMaker ? 'Todo App' : 'Box Maker'}
      </button>
      {showBoxMaker ? <BoxList /> : <TodoList />}
    </div>
  );
}

export default App;
