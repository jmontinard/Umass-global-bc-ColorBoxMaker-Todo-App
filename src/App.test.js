import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import App from './App';

test('renders App component', () => {
  const { container } = render(<App />);
  expect(container).toBeInTheDocument();
});

test('matches snapshot for App component', () => {
  const { container } = render(<App />);
  expect(container).toMatchSnapshot();
});

test('toggles between Box Maker and Todo App', () => {
  const { getByText } = render(<App />);

  // Check initial content
  expect(getByText('Switch to Todo App')).toBeTruthy();
  expect(getByText('Add a new box')).toBeTruthy();

  // Click the button to switch to TodoList
  fireEvent.click(getByText('Switch to Todo App'));

  // Check if the button and TodoList component are rendered
  expect(getByText('Switch to Box Maker')).toBeTruthy();
  expect(getByText('Enter a new todo')).toBeTruthy();
});
