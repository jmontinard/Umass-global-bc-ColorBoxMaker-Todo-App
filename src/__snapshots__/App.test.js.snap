// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for App component 1`] = `
<div>
  <div
    class="App"
  >
    <button>
      Switch to 
      Todo App
    </button>
    <div>
      <form>
        <div>
          <label
            for="width"
          >
            Width:
          </label>
          <input
            id="width"
            name="width"
            type="text"
            value=""
          />
        </div>
        <div>
          <label
            for="height"
          >
            Height:
          </label>
          <input
            id="height"
            name="height"
            type="text"
            value=""
          />
        </div>
        <div>
          <label
            for="backgroundColor"
          >
            Background Color:
          </label>
          <input
            id="backgroundColor"
            name="backgroundColor"
            type="text"
            value=""
          />
        </div>
        <button
          type="submit"
        >
          Add Box
        </button>
      </form>
    </div>
  </div>
</div>
`;
